File: codewars/10-rgb_to_hex/golang/main.go
package main

import (
	"fmt"
	"strings"
)

func RGB(r, g, b int) string {
	ans := ""
	rgb := [...]int{r, g, b}
	hex := strings.Split("0 1 2 3 4 5 6 7 8 9 A B C D E F", " ")

	for _, v := range rgb {
		if v < 0 {
			v = 0
		} else if v > 255 {
			v = 255
		}
		ans += hex[v/16] + hex[v%16]
	}

	return ans
}

// Another solution
func getValid(x int) int {
	switch {
	case x < 0:
		return 0
	case x > 255:
		return 255
	default:
		return x
	}
}

// Another solution
func RGB3(r, g, b int) string {
	return Limit(r) + Limit(g) + Limit(b)
}
func Limit(rgb int) string {
	if rgb < 0 {
		rgb = 0
	}
	if rgb > 255 {
		rgb = 255
	}
	return fmt.Sprintf("%02X", rgb)
}

func RGB2(r, g, b int) string {
	res := fmt.Sprintf("%02X%02X%02X", getValid(r), getValid(g), getValid(b))
	fmt.Println(res)
	return res
}

func main() {
	fmt.Println(RGB(255, 255, 255))
	fmt.Println(RGB(0, 0, 0))
	fmt.Println(RGB(300, -50, 124))
}

File: codewars/10-rgb_to_hex/python/main.py
def rgb(r, g, b):
    ans = ""
    rgb = [r, g, b]
    rgb = [x if 0 <= x <= 255 else (0 if x < 0 else 255) for x in rgb]
    print(rgb)
    hex = "0 1 2 3 4 5 6 7 8 9 A B C D E F".split()
    for i in rgb:
        # if i < 0: i = 0
        # if i > 255: i = 255
        ans += hex[int(i/16)] + hex[int(i%16)]
    return ans

def rgb2(r, g, b):
    round = lambda x: min(255, max(x, 0))
    return ("{:02X}" * 3).format(round(r), round(g), round(b))

def rgb3(r, g, b):
    clamp = lambda x: max(0, min(x, 255))
    return "%02X%02X%02X" % (clamp(r), clamp(g), clamp(b))

def rgb4(r, g, b): 
    return ''.join(['%02X' % max(0, min(x, 255)) for x in [r, g, b]])

if __name__ == "__main__":
    print(rgb(255, 255, 255))
    print(rgb(254, 253, 252))
    print(rgb(-223, 253, 29999))
    print(rgb(0, 0, 0))